helpers/configureImage.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const URLSearchParams = require('url').URLSearchParams;
+const URL = require('url').URL;
+const common = require('./lib/common.js');
+
+const factory = () => {
+    return function(image, options) {
+        if (!common.isObject(image) || !common.isString(image.data) || !common.isValidUrl(image.data)) {
+            // Return empty string if this does not appear to be an image object.
+            return ''
+        }
+
+        // 'https://cdn0.bigcommerce.com/s-hash/images/stencil/{:size}/l/some-image.original.jpg?c=2'
+        let url = new URL(image.data);
+
+        let params = new URLSearchParams(url.search.slice(1));
+
+        //remove AIM query strings if already set by the application
+        params.delete('imbypass');
+        params.delete('imdevicesize');
+        params.delete('imdeviceformat');
+
+        if (options && common.isOptions((options))) {
+            if (options.get('disable-all-optimizations')) {
+                // Add parameter to disable Akamai Image Manager optimization.
+                params.set('imbypass', '1');
+                // Query string is now: 'c=2&imbypass=on'
+            } else {
+                if (options.get('disable-device-sizing')) {
+                    params.set('imdevicesize', '0');
+                } else if (options.get('disable-device-formatting')) {
```

Comment:
You probably want `if`, not `else if` here right?

Reply:
You're right.